### PR TITLE
Add Combine and SwiftUI bridges

### PR DESCRIPTION
**Pull request by moglistree**

### CODE REVIEW

**Review by enhorn — DISMISSED** [review]
Nice work!

**Review by CJEkman — COMMENTED** [review]
Really nice! 👍 I think we can add some more stuff to this though!

File: FlowTests/Signal+CombineTests.swift [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            case .failure(let error):
+                XCTAssertEqual(error as! TestError, .fatal)
+                endExpectation.fulfill()
+            }
+        } receiveValue: { value in
+            XCTFail("Cancelable should have ended")
+        }
+
+        callbacker.callAll(with: .end(TestError.fatal))
+        callbacker.callAll(with: .value(1))
```
Comment (CJEkman) [inline FlowTests/Signal+CombineTests.swift:94]: Tested this locally, and this approach is flaky. It's basically a race condition that only passes some of the time.

Comment (CJEkman) [reply FlowTests/Signal+CombineTests.swift:94]: To clarify, sending `.end(TestError.fatal)` here triggers both lines 25 and 26 in `Signal+Combine.swift`, and whichever happens to finish first will may pass/fail the test:
```Swift
bag += finiteVersion.onError { subscription.end(with: $0) }
bag += finiteVersion.onEnd { subscription.end() }
```

Comment (CJEkman) [reply FlowTests/Signal+CombineTests.swift:93]: Fixed it!

Comment (moglistree) [reply FlowTests/Signal+CombineTests.swift:93]: Cool so we found a race condition

File: Bridges/Signal+Combine.swift [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                    }
+                }
+            }
+        }
+        
+        typealias Output = Value
+        typealias Failure = Error
+        
+        fileprivate var signal: CoreSignal<Kind, Value>
+        fileprivate var bag = DisposeBag()
```
Comment (CJEkman) [inline Bridges/Signal+Combine.swift:41]: Looks like this bag will never be disposed, right? 🤔 Hm...

Comment (CJEkman) [reply Bridges/Signal+Combine.swift]: Maybe it'd be fine, but I'm worried about memory leaks.

Comment (moglistree) [reply Bridges/Signal+Combine.swift]: Good point let me see what I can do

Comment (moglistree) [reply Bridges/Signal+Combine.swift]: In a simple memory leak test:
```
func testMemoryLeaks() {
        let signal = ReadWriteSignal("1")
        let publisher = signal.publisher
        
        let wrappedSignal = publisher.signal
        
        addTeardownBlock { [weak wrappedSignal] in
            XCTAssertNil(wrappedSignal)
        }
    }
```
we do not catch any issues. I think that is why we call `cancel()` in the `deinit` block. My understanding is that this cleans up everything held in the bag. Do you have any other cases in mind?

**Review by enhorn — APPROVED** [review]

**Comment by moglistree** [comment]
Merging this if more changes are need we will tackle them in a separate PR

**Pull request merged**